h2o-algos/src/main/java/hex/generic/Generic.java

```diff
@@ -3,34 +3,30 @@
 import hex.ModelBuilder;
 import hex.ModelCategory;
 import hex.genmodel.*;
-import hex.genmodel.algos.drf.DrfMojoModel;
-import hex.genmodel.algos.gbm.GbmMojoModel;
-import hex.genmodel.algos.glm.GlmMojoModel;
-import hex.genmodel.algos.glm.GlmMultinomialMojoModel;
-import hex.genmodel.algos.glm.GlmOrdinalMojoModel;
-import hex.genmodel.algos.isofor.IsolationForestMojoModel;
-import hex.genmodel.algos.kmeans.KMeansMojoModel;
-import water.DKV;
 import water.H2O;
 import water.Key;
-import water.Scope;
 import water.fvec.ByteVec;
 import water.fvec.Frame;
-import water.util.ArrayUtils;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Objects;
+import java.util.*;
 
 /**
  * Generic model able to do scoring with any underlying model deserializable into a format known by the {@link GenericModel}.
  * Only H2O Mojos are currently supported.
  */
 public class Generic extends ModelBuilder<GenericModel, GenericModelParameters, GenericModelOutput> {
+    
+    private static final Set<String> ALLOWED_MOJO_ALGOS;
+    static{
```

Comment:
We have tests that prove all these algorithms are supported and work. Both JUnit (`GenericModelTest`) and Python and R (pyunit_generic* or runit_generic*).

There weas a **false support** claimed for KMeans, but in fact it was not inteded, nor tested in any way. Probably a forgotten leftover. As this section had to be redesigned in this PR, took the opportunity to remove the support. WDYT ? cc @michalkurka